src/libopensc/pkcs15.c

```diff
@@ -2391,6 +2395,23 @@ sc_pkcs15_parse_unusedspace(const unsigned char *buf, size_t buflen, struct sc_p
 	return 0;
 }
 
+static int decompress_file(sc_card_t *card, unsigned char *buf, size_t buflen, 
+		unsigned char **out, size_t *outlen)
+{
+	LOG_FUNC_CALLED(card->ctx);
+#ifdef ENABLE_ZLIB
+	int rv = SC_SUCCESS;
+
+	rv = sc_decompress_alloc(out, outlen, buf, buflen, COMPRESSION_AUTO);
+	if (rv != SC_SUCCESS) {
+		sc_log(card->ctx,  "Decompression failed: %d", rv);
+		LOG_FUNC_RETURN(card->ctx, SC_ERROR_INVALID_DATA);
+	}
+	LOG_FUNC_RETURN(card->ctx, SC_SUCCESS);
+#endif
+	sc_log(card->ctx, "Compression not supported, no zlib");
+	LOG_FUNC_RETURN(card->ctx, SC_ERROR_NOT_SUPPORTED);
```

Comment:
please enclose the return of `SC_ERROR_NOT_SUPPORTED` in `#else`/`#endif`, otherwise static code analysis will complain that the last part is not reachable